src/main/java/com/mongodb/jdbc/logging/QueryDiagnostics.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.jdbc.logging;
+
+import static com.mongodb.jdbc.utils.BsonUtils.JSON_WRITER_NO_INDENT_SETTINGS;
+
+import com.mongodb.jdbc.MongoDriver;
+import com.mongodb.jdbc.MongoJsonSchema;
+import com.mongodb.jdbc.utils.BsonUtils;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.codecs.Codec;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+public class QueryDiagnostics {
+    private static final Codec<QueryDiagnostics> CODEC =
+            MongoDriver.getCodecRegistry().get(QueryDiagnostics.class);
+
+    @BsonProperty private String sqlQuery;
+    @BsonProperty private BsonDocument namespacesSchema;
```

Comment:
If we decide to go with "catalog" instead of "namespacesSchema", I assume we'll update the references in these other files as well.